src/channels/views/ChannelCreate/useSaveChannel.ts

```diff
@@ -0,0 +1,66 @@
+import { MutationFunction } from "@apollo/client";
+import { useUser } from "@dashboard/auth";
+import { ChannelWarehouses } from "@dashboard/channels/pages/ChannelDetailsPage/types";
+import {
+  ChannelCreateInput,
+  ChannelCreateMutation,
+  ChannelReorderWarehousesMutation,
+  Exact,
+  ReorderInput,
+} from "@dashboard/graphql";
+import { extractMutationErrors } from "@dashboard/misc";
+
+import { calculateItemsOrderMoves } from "../ChannelDetails/handlers";
+
+interface SaveChannelConfig {
+  createChannel: MutationFunction<
+    ChannelCreateMutation,
+    Exact<{ input: ChannelCreateInput }>
+  >;
+  reorderChannelWarehouses: MutationFunction<
+    ChannelReorderWarehousesMutation,
+    Exact<{
+      channelId: string;
+      moves: ReorderInput | ReorderInput[];
+    }>
+  >;
+}
+
+export const useSaveChannel = ({
+  createChannel,
+  reorderChannelWarehouses,
+}: SaveChannelConfig) => {
+  const { refetchUser } = useUser();
+
+  return async (
+    input: ChannelCreateInput,
+    warehousesToDisplay: ChannelWarehouses,
+  ) => {
+    const createChannelMutation = createChannel({
+      variables: {
+        input,
+      },
+    });
+
+    const result = await createChannelMutation;
```

Comment:
question: should you fire this function or createChannel return promise
` const result = await createChannelMutation()`;

Reply:
extractMutationErrors requires promise - i used existing code